File: fortran/tests/run_all_tests.f90
! Master test runner for all Fortran algorithm modules
program run_all_tests
    use iso_fortran_env, only: int32
    implicit none

    integer :: total_modules = 8
    integer :: passed_modules = 0
    integer :: exit_code

    print '(A)', ""
    print '(A)', "============================================"
    print '(A)', "   Algorithms Multiverse - Fortran"
    print '(A)', "        Complete Test Suite"
    print '(A)', "============================================"
    print '(A)', ""

    ! Run each module's test suite
    call run_test("Graph Module", "./graph_module_test", passed_modules)
    call run_test("String Module", "./string_module_test", passed_modules)
    call run_test("Crypto Module", "./crypto_module_test", passed_modules)
    call run_test("Geometry Module", "./geometry_module_test", passed_modules)
    call run_test("Data Structures Module", "./data_structures_module_test", passed_modules)
    call run_test("Statistics Module", "./statistics_module_test", passed_modules)
    call run_test("Optimization Module", "./optimization_module_test", passed_modules)
    call run_test("Streaming Module", "./streaming_module_test", passed_modules)

    ! Print summary
    print '(A)', ""
    print '(A)', "============================================"
    print '(A)', "           TEST SUITE SUMMARY"
    print '(A)', "============================================"
    print '(A,I0,A,I0,A)', "Modules Passed: ", passed_modules, "/", total_modules

    if (passed_modules == total_modules) then
        print '(A)', ""
        print '(A)', "   ✓✓✓ ALL TESTS PASSED ✓✓✓"
        print '(A)', ""
    else
        print '(A)', ""
        print '(A)', "   ✗✗✗ SOME TESTS FAILED ✗✗✗"
        print '(A)', ""
    end if
    print '(A)', "============================================"

contains

    subroutine run_test(module_name, executable, passed_count)
        character(len=*), intent(in) :: module_name, executable
        integer, intent(inout) :: passed_count
        integer :: exit_status
        character(len=256) :: command

        print '(A,A,A)', "Running ", module_name, " tests..."

        command = trim(executable)
        call execute_command_line(command, exitstat=exit_status)

        if (exit_status == 0) then
            print '(A,A,A)', "✓ ", module_name, " tests completed successfully"
            passed_count = passed_count + 1
        else
            print '(A,A,A)', "✗ ", module_name, " tests failed or could not run"
        end if

        print '(A)', ""

    end subroutine run_test

end program run_all_tests

File: javascript/tests/runAllTests.js
#!/usr/bin/env node

/**
 * Test runner for all algorithm modules
 * Run this file to execute all tests
 */

import { execSync } from 'child_process';
import { readdirSync } from 'fs';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));

console.log('====================================');
console.log('  ALGORITHMS MULTIVERSE TEST SUITE  ');
console.log('====================================\n');

// Get all test files
const testFiles = readdirSync(__dirname)
    .filter(file => file.endsWith('.test.js'))
    .sort();

let totalTests = 0;
let passedTests = 0;
let failedTests = [];

// Run each test file
for (const testFile of testFiles) {
    console.log(`\n📝 Running ${testFile}...`);
    console.log('─'.repeat(40));

    try {
        execSync(`node ${join(__dirname, testFile)}`, {
            stdio: 'inherit',
            cwd: __dirname
        });
        passedTests++;
        console.log(`✅ ${testFile} completed successfully`);
    } catch (error) {
        failedTests.push(testFile);
        console.error(`❌ ${testFile} failed`);
    }

    totalTests++;
}

// Summary
console.log('\n' + '='.repeat(50));
console.log('                TEST SUMMARY');
console.log('='.repeat(50));
console.log(`Total test files: ${totalTests}`);
console.log(`✅ Passed: ${passedTests}`);
console.log(`❌ Failed: ${failedTests.length}`);

if (failedTests.length > 0) {
    console.log('\nFailed tests:');
    failedTests.forEach(test => console.log(`  - ${test}`));
    process.exit(1);
} else {
    console.log('\n🎉 All tests passed successfully!');
    process.exit(0);
}